goci-interfaces/goci-ui/src/main/java/uk/ac/ebi/spot/goci/ui/service/impl/GeneImpcLinkServiceImpl.java

```diff
@@ -0,0 +1,56 @@
+package uk.ac.ebi.spot.goci.ui.service.impl;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.MediaType;
+import org.springframework.stereotype.Service;
+import org.springframework.util.StreamUtils;
+import org.springframework.web.client.RequestCallback;
+import org.springframework.web.client.RestTemplate;
+import uk.ac.ebi.spot.goci.ui.component.ImpcIdentifierMap;
+import uk.ac.ebi.spot.goci.ui.service.GeneImpcLinkService;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.util.Arrays;
+import java.util.Map;
+
+@Service
+public class GeneImpcLinkServiceImpl implements GeneImpcLinkService {
+
+    private static final Logger log = LoggerFactory.getLogger(GeneImpcLinkServiceImpl.class);
+
+    @Autowired
+    ImpcIdentifierMap impcIdentifierMap;
+
+    @Value("${impc.bukupload.uri}")
+    private String endpoint;
+
+    @Autowired
+    RestTemplate restTemplate;
+
+    /** Service makes a call to Impc File based API & extracts the content of file in to a HashMap
+     *
+     */
+    public void populateGeneImpcMap() {
+        RequestCallback callback = clientHttpRequest -> clientHttpRequest.getHeaders()
+                .setAccept(Arrays.asList(MediaType.APPLICATION_OCTET_STREAM));
```

Comment:
I think Collections.singletonList(..) will remove the code smell here for a one-element-list [https://jsparrow.github.io/rules/use-collections-singleton-list.html#description]